txbits/app/wallet/Wallet.scala

```diff
@@ -304,6 +358,7 @@ class Wallet(rpc: JsonRpcHttpClient, currency: CryptoCurrency, nodeId: Int, para
   }
 
   private def getBalance = {
+    // This method is unused as balance is now tracked in the database
```

Comment:
No, if it has less funds than needed but thought it had enough for a withdrawal it was trying to send, it would try to send and an exception would be thrown. This would clear the wallet state and it would try again next block time. It would keep doing this until it sends. In the meantime, deposits would continue to be processed but every deposit would be checked against the database as it would have no cache to tell whether or not it's been confirmed.